applications/nrf5340_audio/src/audio/sw_codec_select.c

```diff
@@ -115,37 +203,74 @@ int sw_codec_decode(uint8_t const *const encoded_data, size_t encoded_size, bool
 	}
 
 	int ret;
-	char pcm_data_mono[PCM_NUM_BYTES_MONO] = {0};
+
 	static char pcm_data_stereo[PCM_NUM_BYTES_STEREO];
 
 	size_t pcm_size_stereo = 0;
-	uint16_t pcm_size_session = 0;
+	size_t pcm_size_mono = 0;
+	size_t decoded_data_size = 0;
 
 	switch (m_config.sw_codec) {
 	case SW_CODEC_LC3: {
 #if (CONFIG_SW_CODEC_LC3)
-		/* Typically used for right channel if stereo signal */
-		char pcm_data_mono_right[PCM_NUM_BYTES_MONO] = {0};
-
 		switch (m_config.decoder.channel_mode) {
 		case SW_CODEC_MONO: {
+			char *pcm_data_mono_left_ptr;
+
+			char pcm_data_mono_left[PCM_NUM_BYTES_MONO] = {0};
+			char decoded_data_mono_left[PCM_NUM_BYTES_MONO] = {0};
+
+#ifdef CONFIG_AUDIO_SAMPLE_RATE_CONVERSION
+			char decoded_data_mono_left_converted[PCM_NUM_BYTES_MONO] = {0};
+#endif /* CONFIG_AUDIO_SAMPLE_RATE_CONVERSION */
+
 			if (bad_frame && IS_ENABLED(CONFIG_SW_CODEC_OVERRIDE_PLC)) {
-				memset(pcm_data_mono, 0, PCM_NUM_BYTES_MONO);
-				pcm_size_session = PCM_NUM_BYTES_MONO;
+				memset(pcm_data_mono_left, 0, PCM_NUM_BYTES_MONO);
+				pcm_size_mono = PCM_NUM_BYTES_MONO;
 			} else {
-				ret = sw_codec_lc3_dec_run(encoded_data, encoded_size,
-							   LC3_PCM_NUM_BYTES_MONO, 0, pcm_data_mono,
-							   &pcm_size_session, bad_frame);
+				ret = sw_codec_lc3_dec_run(
+					encoded_data, encoded_size, LC3_PCM_NUM_BYTES_MONO, 0,
+					decoded_data_mono_left, (uint16_t *)&decoded_data_size,
+					bad_frame);
 				if (ret) {
 					return ret;
 				}
+
+				if (m_config.decoder.sample_rate_hz ==
+				    CONFIG_AUDIO_SAMPLE_RATE_HZ) {
+					pcm_data_mono_left_ptr = decoded_data_mono_left;
+					pcm_size_mono = decoded_data_size;
+				} else {
+#ifdef CONFIG_AUDIO_SAMPLE_RATE_CONVERSION
+					ret = sample_rate_converter_process(
+						&sample_rate_converter_left,
+						SAMPLE_RATE_FILTER_TEST,
+						(void *)decoded_data_mono_left, decoded_data_size,
+						m_config.decoder.sample_rate_hz,
+						(void *)decoded_data_mono_left_converted,
+						ARRAY_SIZE(decoded_data_mono_left_converted),
+						&pcm_size_mono, CONFIG_AUDIO_SAMPLE_RATE_HZ);
+					if (ret) {
+						LOG_ERR("Sample rate conversion "
+							"failed for left "
+							"channel: "
+							"%d",
+							ret);
+						return ret;
+					}
```

Comment:
Is this due to clang? We should see if we can reduce the number of indents here.
Instead of doing if/else, maybe we can do a `continue` in the `if` and then drop the else indent.

Reply:
Discussed: Will hopefully be solved by all the other changes

Reply:
This got a bit better with the re-writes, but it's still not great. I think your idea to break in the `bad_frame` would work and can also reduce some calculations as we in those cases can skip `pscm_zero_pad` and `pscm_combine`, but as that's a bigger re-write I suggest we do that in a separate commit and PR.